### ISSUE TITLE
MinGW: add -DLIBDEFLATE_DLL to pkg-config file

### ISSUE THREAD

**Issue by vtorri** [ebiggers/libdeflate#229]
-DLIBDEFLATE_DLL should be added to CFlags
Also, add -ULIBDEFLATE_DLL to CFlags.private (this is for static linking)

**Comment by kmilos** [comment]
FYI: already done for MSYS2 packages for a while: https://github.com/msys2/MINGW-packages/tree/master/mingw-w64-libdeflate

**Comment by vtorri** [comment]
@kmilos FYI: I know, as my original comment is based on it...

but i think it is better to fix things upstream rather than having patches everywhere...

**Comment by kmilos** [comment]
This can't really be fixed upstream as there is no build system used currently as you know (#101, #191) - you don't want to have these on platforms other than Windows...

**Comment by ebiggers** [comment]
Fixed by commit 8a26b31bf80389c22c5559d56cb90fe60d2327b7.

> This can't really be fixed (elegantly ) upstream as there is no build system used currently as you know (#101, #191) - you don't want to have these on platforms other than Windows...

Of course there is a build system, just not the one that some people like.  I am looking into switching it to CMake.

Anyway, that is irrelevant here since CMake doesn't have any built-in support for pkg-config, so generating `libdeflate.pc` using CMake would be nearly identical to what the plain Makefile does currently.  So I just fixed the pkg-config file issue now.
